WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt

```diff
@@ -150,6 +159,24 @@ class JetpackMigrationViewModel @Inject constructor(
         )
     }
 
+    private fun extractLanguageFromFlagsMap(userPrefs: Map<String, Any?>): String {
+        val languageKey = localeManagerWrapper.getLocalePrefKeyString()
+        return userPrefs[languageKey] as? String ?: ""
+    }
+
+    private fun emitLanguageRefreshIfNeeded(languageCode: String) {
+        if (languageCode.isNotEmpty()) {
+            val shouldEmitLanguageRefresh = !localeManagerWrapper.isSameLanguage(languageCode)
+            if (shouldEmitLanguageRefresh) {
+                _refreshAppLanguage.value = languageCode
+            }
+        }
+    }
```

Comment:
Not suggesting a change here, but wanted to note that my first thought here was if we could instead use:

```kotlin
val refreshAppLanguage: LiveData<String> = _refreshAppLanguage.distinct()
```

but I guess this could result in one unnecessary refresh at the start compared to the approach used here, since this way compares to the locale set in the current context. :man_shrugging:

Reply:
In the future, it would be nice if this idempotency was intrinsic to the implementation, so we wouldn't need to worry about it at the call site, but that's out of scope for this PR :sweat_smile: .

Reply:
> In the future, it would be nice if this idempotency was intrinsic to the implementation, so we wouldn't need to worry about it at the call site

Top idea 👏 , totally agreeing with you on this one, that would add to the reusability of the code 🚀 

I've added a task to our team's backlog to further pursue this when there's time for technical improvements 🙇 . On the other hand, it's possible we won't need this at all if we add support for the Android 13 [Per-app language preferences](https://developer.android.com/guide/topics/resources/app-languages) mechanism soon 🤔

Reply:
Good point, this could all become moot in the future :smile: :+1: